### PR TITLE
Changes as discussed in issue #388

### PR DESCRIPTION
**Pull request by arunwizz**
@tipsy - this is to handle issue discussed under #388. I am not sure where to add unit cases. So please review and advice. 

Regards,
Arun

### CODE REVIEW

**Comment by tipsy** [comment]
@perwendel could you have a look at this?

**Comment by arunwizz** [comment]
@perwendel did you get a chance to review this PR?

**Comment by tipsy** [comment]
Apparently he didn't. Do you still want this merged @arunwizz ?

**Comment by arunwizz** [comment]
@tipsy sorry, I forgot about it, i's upto you guys.. If you feel the PR is worth can pull it else please feel free to close it.

**Comment by tipsy** [comment]
I think it's a good change, @arunwizz. If you update the PR and change method names from `setInitExceptionHandler` to just `initExceptionHandler`, I will merge it.

You should add a 
```java
if (initialized) {
    throwBeforeRouteMappingException();
}
```
Too, I guess.

**Comment by arunwizz** [comment]
Thanks @tipsy! I have incorporated changes as suggested. Let me know if any more issues.

**Comment by arunwizz** [comment]
@tipsy - did you get a chance to look at this PR? Updated as discussed.

**Comment by tipsy** [comment]
Sorry, got lost. Give me a sec.

File: src/main/java/spark/Service.java [hunk 1]
```diff
@@ -579,6 +590,20 @@ public HaltException halt(int status, String body) {
     }
 
     /**
+     * Overrides default exception handler during initialization phase
+     *
+     * @param initExceptionHandler
+     *          The custom init exception handler
+     */
+    public void initExceptionHandler(
+        Consumer<Exception> igniteExceptionHandler) {
```
Comment (tipsy) [inline src/main/java/spark/Service.java]: Move up to previous line

File: src/main/java/spark/Spark.java [hunk 2]
```diff
@@ -1031,6 +1032,17 @@ public static void secure(String keystoreFile,
     }
 
     /**
+     * Overrides default exception handler during initialization phase
+     * 
+     * @param initExceptionHandler
+     *          The custom init exception handler
+     */
+    public static void initExceptionHandler(
+        Consumer<Exception> igniteExceptionHandler) {
```
Comment (tipsy) [inline src/main/java/spark/Spark.java]: Move up

**Comment by tipsy** [comment]
Can you just move up these two lines?

**Comment by tipsy** [comment]
@arunwizz, I merged it and pushed the formatting fixes. Thanks for the PR!

**Pull request merged**